libraries/AC_AttitudeControl/AC_AttitudeControl.cpp

```diff
@@ -614,7 +614,8 @@ void AC_AttitudeControl::input_thrust_vector_heading(const Vector3f& thrust_vect
         _ang_vel_target.z = input_shaping_angle(attitude_error.z, _input_tc, get_accel_yaw_max_radss(), _ang_vel_target.z, heading_rate, get_slew_yaw_rads(), _dt);
 
         // Limit the angular velocity
-        ang_vel_limit(_ang_vel_target, radians(_ang_vel_roll_max), radians(_ang_vel_pitch_max), MIN(radians(_ang_vel_yaw_max), get_slew_yaw_rads()));
+        const float ang_vel_yaw_max = is_zero(_ang_vel_yaw_max)?get_slew_yaw_rads():MIN(radians(_ang_vel_yaw_max), get_slew_yaw_rads());
+        ang_vel_limit(_ang_vel_target, radians(_ang_vel_roll_max), radians(_ang_vel_pitch_max), ang_vel_yaw_max);
```

Comment:
```suggestion
        // a zero _angle_vel_yaw_max means that setting is disabled
        const float ang_vel_yaw_max_rads = is_zero(_ang_vel_yaw_max) ? get_slew_yaw_rads() : MIN(radians(_ang_vel_yaw_max), get_slew_yaw_rads());
        ang_vel_limit(_ang_vel_target, radians(_ang_vel_roll_max), radians(_ang_vel_pitch_max), ang_vel_yaw_max_rads);
```

This isn't a blocker, but whitespace is cheap, even with Covid-related supply-chain issues.

Tacking rads to me seems to make sense given what the last line looks like.

Reply:
I think we need to move this to the top of the function and also use it to limit the initial yaw rate command on line 593.

Reply:
Moved to top as requested.